_studio/shared/src/libmfx_core_vaapi.cpp

```diff
@@ -612,54 +612,35 @@ VAAPIVideoCORE::AllocFrames(
         } else
         {
             // external allocator
-            if (m_bSetExtFrameAlloc && request->Info.FourCC != MFX_FOURCC_P8)
+            if (m_bSetExtFrameAlloc &&
+                request->Info.FourCC != MFX_FOURCC_P8 &&
+                (request->Type & MFX_MEMTYPE_EXTERNAL_FRAME))
             {
+                // make 'fake' Alloc call to retrieve memId's of surfaces already allocated by app.
                 sts = (*m_FrameAllocator.frameAllocator.Alloc)(m_FrameAllocator.frameAllocator.pthis, &temp_request, response);
 
-                // if external allocator cannot allocate d3d frames - use default memory allocator
-                if (MFX_ERR_UNSUPPORTED == sts || MFX_ERR_MEMORY_ALLOC == sts)
-                {
-                    // Default Allocator is used for internal memory allocation only
-                    if (request->Type & MFX_MEMTYPE_EXTERNAL_FRAME)
-                        return sts;
-                    m_bUseExtAllocForHWFrames = false;
-                    sts = DefaultAllocFrames(request, response);
-                    MFX_CHECK_STS(sts);
+                m_bUseExtAllocForHWFrames = false;
```

Comment:
Is it needed?

Reply:
I think yes, because earlier, when sts! = mfx_err_none we set this variable to false /